src/pages/SwapList/LedgerActionDialogBody.tsx

```diff
@@ -8,18 +8,40 @@ import {
 import React from "react";
 import { LedgerAction } from "../../api/getAction";
 import { toMainUnit } from "../../api/unit";
+import InfoMessage from "../../components/InfoMessage";
 import Web3SendTransactionButton from "../../components/Web3SendTransactionButton";
 import CopyToClipboardButton from "./CopyToClipboard";
 
 interface LedgerActionDialogBodyProps {
-  onClose: () => void;
   action: LedgerAction;
+  onSuccess: (transactionId?: string) => void;
+  onClose: () => void;
+  actionDoneBefore?: boolean;
+}
+
+function actionDoneMessage() {
+  return (
+    <InfoMessage
+      text="You have already performed this action. Do not try again unless you are sure it didn't go through last time."
```

Comment:
```suggestion
      text="You have already performed this action. Do not try again unless you are sure it failed last time."
```
Just because it's shorter